ubg/src/mafia/MafiaNight.js

```diff
@@ -1,12 +1,116 @@
-import React from 'react';
+import React, {useEffect} from 'react';
+import {useParams} from 'react-router-dom';
+import {
+  useFirestore,
+  useUser,
+  useFirestoreCollectionData,
+} from 'reactfire';
+import {makeStyles} from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import Player from './Player';
+import PersonalInfo from './PersonalInfo';
+
+const useStyles = makeStyles((theme) => ({
+  main: {
+    height: '100vh',
+  },
+  video: {
+    background: theme.palette.primary.main,
+    textAlign: '-webkit-center',
+  },
+  gameContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'relative',
+  },
+  card: {
+    width: '100%',
+  },
+}));
 
 /**
  * @return {ReactElement} Mafia night element
  */
 export default function MafiaNight() {
+  const classes = useStyles();
+  const user = useUser();
+  const [initialize, setInitialize] = React.useState(false);
+  const [players, setPlayers] = React.useState([]);
+  const [userInfo, setUserInfo] = React.useState('');
+  const [roleText, setRoleText] = React.useState('');
+  const {roomId} = useParams();
+  const room = useFirestore().collection('rooms').doc(roomId);
+  const usersCollection = room.collection('users');
+  const usersData = useFirestoreCollectionData(usersCollection);
```

Comment:
I'm not sure if we should have another listener here, since we are already listening to this collection in Room, we could pass that prop all the way to this component, instead of adding another listener, but what do you think?

Reply:
The same would go for useUser probably, but idk what the best option is.